hamilton/caching/stores/memory.py

```diff
@@ -0,0 +1,279 @@
+
+        :param result_store: ResultStore implementation to use for persistence.
+            If None, a FileResultStore is created with the default path "./.hamilton_cache".
+        """
+        if result_store is None:
+            result_store = FileResultStore(path="./.hamilton_cache")
+
+        for data_version, stored_result in self._results.items():
+            result_store.set(data_version, stored_result.value)
+
+    @classmethod
+    def load_from(
+        cls,
+        result_store: ResultStore,
+        metadata_store: Optional[MetadataStore] = None,
+        data_versions: Optional[Sequence[str]] = None,
+    ) -> "InMemoryResultStore":
+        """Load in-memory results from another ResultStore instance.
+
+        Since result stores do not store an index of their keys, you must provide a
+        ``MetadataStore`` instance or a list of ``data_version`` for which results
+        should be loaded in memory.
+
+        :param result_store: ``ResultStore`` instance to load results from.
+        :param metadata_store: ``MetadataStore`` instance from which all ``data_version`` are retrieved.
+        :return: InMemoryResultStore copy of the ``result_store``.
+
+        .. code-block:: python
+
+            from hamilton import driver
+            from hamilton.caching.stores.sqlite import SQLiteMetadataStore
+            from hamilton.caching.stores.memory import InMemoryMetadataStore
+            import my_dataflow
+
+            sqlite_metadata_store = SQLiteMetadataStore(path="./.hamilton_cache")
+            in_memory_metadata_store = InMemoryMetadataStore.load_from(sqlite_metadata_store)
+
+            # create the Driver with the in-memory metadata store
+            dr = (
+              driver.Builder()
+              .with_modules(my_dataflow)
+              .with_cache(metadata_store=in_memory_metadata_store)
+              .build()
+            )
+
+
+        """
+        if metadata_store is None and data_versions is None:
+            raise ValueError(
+                "A `metadata_store` or `data_versions` must be provided to load results."
+            )
+
+        in_memory_result_store = InMemoryResultStore()
+
+        data_versions_to_retrieve = set()
+        if data_versions is not None:
+            data_versions_to_retrieve.update(data_versions)
+
+        if metadata_store is not None:
+            for run_id in metadata_store.get_run_ids():
+                for node_metadata in metadata_store.get_run(run_id):
+                    data_versions_to_retrieve.add(node_metadata["data_version"])
+
+        for data_version in data_versions_to_retrieve:
+            # TODO disambiguate "result is None" from the sentinel value when `data_version`
+            # is not found in `result_store`.
+            result = result_store.get(data_version)
+            in_memory_result_store.set(data_version, result)
+
+        # TODO could be beneficial to delete the `result_store` after loading from it,
```

Comment:
`del` would just be a reference clear unless you wipe it, python does not have pointer capabilities in the normal use. Just document how to keep things clean IMO

Reply:
`del result_store` doesn't do much -- so yeah, remove it